src/ecore.js

```diff
@@ -861,6 +868,9 @@ EClass.values = {
 
         return _.isArray(subTypes) ? subTypes : [];
     },
+    hasSubTypes: function () {
```

Comment:
Agreed the inline logic appears to be doing most of this, so we just simplified it by doing the check for xsi:type or a mismatch in the featureType and the class type.